### PR TITLE
Split up sleepy head into four separate recipes

### PR DESCRIPTION
**Pull request by codybsauer**
New recipes match the old assignments in spirit, but get into more
details about how booleans can be used and provide a gradual progression
of topics.

-Sleepy head uses a boolean in an if-else statement
-Star student uses a boolean in a method and introduces the not operator
-Game over uses a boolean in a while loop
-Red Square uses multiple booleans with and/or operators

### CODE REVIEW

**Review by ColbySchexnayder — CHANGES_REQUESTED** [review]
I like this a lot better than what we had. The package names jump from 4 to 6 at googly eyes though

File: Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		// if either boolean is true rather than only if both are true.
+		
+		// 7. Write an if statement using the static booleans at the top of 
+		//    the program. If drewRedSquare OR drewBlueSquare are true, then
+		//    tell the user they drew a square in a pop-up. Hint: ||
+		
+		// 8. Write an else-if statement that tells the user they drew a 
+		//    triangle if drewRedTriangle or drewBlueTriangle are true.
+		
+		// 9. Keep rerunning the program and change the isRed and isSquare
```
Comment (tylercwiltz) [inline Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java:55]: typo change "step 5" to "step 6"

File: Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+		
+		// 6. Keep rerunning your program, initializing isRed and IsSquare to 
+		//    different values until you've seen a red square, red triangle, 
+		//    blue square and blue triangle drawn.
+		
+		// Much like the "and" operator, the "or" operator || can be used to 
+		// combine two boolean statements. However the statement becomes true
+		// if either boolean is true rather than only if both are true.
+		
+		// 7. Write an if statement using the static booleans at the top of
```
Comment (tylercwiltz) [inline Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java:48]: Maybe mention something about how those booleans are set in the draw methods below? Just so it's clear to the students that calling that method could change the value of the boolean from what it is initialized to at the top of the code.

**Review by tylercwiltz — COMMENTED** [review]
Just a few suggestions and a few typos I noticed. Overall, I think this is a more practical use of booleans than what we had and splitting it up will hopefully be less confusing/more enjoyable for the students.

File: Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+		
+		// 1. Run the program and notice drawRedSquare is not called.
+		
+		// 2. Initialize isRed to true instead of false and run the program 
+		//    again.
+		
+		// 3. Write an else if statement that calls drawBlueSquare if isRed is
+		//    false AND isSquare is true. Hint: Use ! and &&.
+		
+		// 4. Write an else if statement that calls drawRedTriangle() is isRed
```
[the remark this answers is not in the document]
Comment (codybsauer) [reply Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java:34]: fix'd

File: Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+		//    is true and isSquare is false.
+		
+		// 5. Write an else if statement that calls drawBlueTriangle if both
+		//    isRed and isSquare are false.
+		
+		// 6. Keep rerunning your program, initializing isRed and IsSquare to
```
[the remark this answers is not in the document]
Comment (codybsauer) [reply Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java:40]: fixed

File: Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java (hunk 1 above)
Comment (codybsauer) [reply Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java:55]: fixed

File: Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java (hunk 2 above)
Comment (codybsauer) [reply Level0-Module4/src/_02_boolean/_4_red_square/RedSquare.java:48]: Added:
		//    Notice these static booleans start off false, but update to true 
		//    whenever you call their matching method.

**Review by tylercwiltz — APPROVED** [review]

**Comment by jaLeague** [comment]
I like the changes. 
Minor comment. RedSquare #1 says "notice drawRedSquare is not called". The student does not know if it is not called or the program just isn't working. It would be nice to have an else that confirmed program ran but did not draw anything. The else statement could stay around even after the else if's are added later. Always good to encourage an else statement any way :)

**Review by keithagroves — APPROVED** [review]

**Pull request merged**